state/application.go

```diff
@@ -2126,32 +2126,30 @@ func (a *Application) removeUnitOps(u *Unit, asserts bson.D, op *ForcedOperation
 	ops = append(ops, hostOps...)
 
 	m, err := a.st.Model()
-	if op.FatalError(err) {
+	if err != nil {
 		return nil, errors.Trace(err)
-	} else {
-		if m.Type() == ModelTypeCAAS {
-			ops = append(ops, u.removeCloudContainerOps()...)
-		}
-		branchOps, err := unassignUnitFromBranchOp(u.doc.Name, a.doc.Name, m)
-		if err != nil {
-			if !op.Force {
-				return nil, errors.Trace(err)
-			}
-			op.AddError(err)
+	}
+	if m.Type() == ModelTypeCAAS {
+		ops = append(ops, u.removeCloudContainerOps()...)
+	}
+	branchOps, err := unassignUnitFromBranchOp(u.doc.Name, a.doc.Name, m)
+	if err != nil {
+		if !op.Force {
+			return nil, errors.Trace(err)
 		}
-		ops = append(ops, branchOps...)
+		op.AddError(err)
 	}
+	ops = append(ops, branchOps...)
 
 	sb, err := NewStorageBackend(a.st)
+	if err != nil {
```

Comment:
Similarly we could still do the charm and application cleanup further down even if getting the storage backend failed.

Reply:
This is the same issue as above - NewStorageBackend() only fails if st.Model() fails.